refract-spec.md

```diff
@@ -12,89 +12,279 @@ This document conforms to RFC 2119, which says:
 
 [MSON](https://github.com/apiaryio/mson) is used throughout this document to describe and define data structures.
 
-## Data Structure
+## Primitive Types
 
-Refract provides a single structure for data, which will be referred to throughout this document as an element. All elements found in a Refract document SHOULD conform to the element structure as defined here.
+Primitive Types are types used by Refract as the basis for all values and structure. These types use both JSON primitive values along with Refract Elements.
```

Comment:
In other words, refract uses these "Types" to describe other structures. If that is what you are saying, I think you need to make that crystal clear and have some examples (probably around objects) to make clear the differentiation.